plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/variation-selector/frontend.ts

```diff
@@ -232,30 +320,52 @@ const { actions, state } = store< VariableProductAddToCartWithOptionsStore >(
 				}
 			},
 			handlePillClick() {
-				if ( state.isOptionDisabled ) {
-					return;
-				}
 				const context = getContext< Context >();
+
 				if ( context.selectedValue === context.option.value ) {
 					context.selectedValue = '';
 				} else {
 					context.selectedValue = context.option.value;
 				}
 				actions.setAttribute( context.name, context.selectedValue );
+				if ( inAutoselectScope ) {
+					// Avoid infinite recursion
+				} else {
+					attributesAutoselectOthers();
+				}
 			},
 			handleDropdownChange( event: ChangeEvent< HTMLSelectElement > ) {
 				const context = getContext< Context >();
 				context.selectedValue = event.currentTarget.value;
 				actions.setAttribute( context.name, context.selectedValue );
+				if ( inAutoselectScope ) {
+					// Avoid infinite recursion
+				} else {
+					attributesAutoselectOthers();
+				}
 			},
 		},
 		callbacks: {
 			setDefaultSelectedAttribute() {
-				const context = getContext< Context >();
+				const context = getContext< Context >(),
+					form = document.querySelector( 'form.wc-block-add-to-cart-with-options' ),
+					variation_selectors = form.querySelectorAll( '.wp-block-woocommerce-add-to-cart-with-options-variation-selector-attribute-options' ),
+					autoselect =
+						variation_selectors[0].dataset?.autoselect ||
+						false;
```

Comment:
There might be multiple _Add to Cart + Options_ blocks on the same screen, which might make this code fail. Instead, we could pass `autoselect` as part of the Interactivity API context, which will couple it to this instance of the block. You should be able to add it here:

https://github.com/woocommerce/woocommerce/blob/fae4f1968f21071e6acd0e8643f1e97473bb5b63/plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptions/VariationSelectorAttributeOptions.php#L190-L195

And then, in the frontend script you can get it directly from the `context`:

```TS
const context = getContext< Context >();
const { autoselect } = context;
```